tests/test_clamp.py

```diff
@@ -70,3 +70,114 @@ def test_clamp_and_stimulate_api():
 
     vs2 = jnp.concatenate([vs21, vs22])
     assert np.max(np.abs(vs1 - vs2)) < 1e-8
+
+
+def test_data_clamp():
+    """Data clamp with no stimuli or data_stimuli, and no t_max (should get defined by the clamp)."""
+    comp = jx.Compartment()
+    comp.insert(HH())
+    comp.record()
+    clamp = -50.0 * jnp.ones((1000,))
+
+    def provide_data(clamp):
+        data_clamps = comp.data_clamp("v", clamp)
+        return data_clamps
+
+    def simulate(clamp):
+        data_clamps = provide_data(clamp)
+        return jx.integrate(comp, data_clamps=data_clamps)
+
+    jitted_simulate = jax.jit(simulate)
+
+    s = jitted_simulate(clamp)
+    assert np.all(s[:, 1:] == -50.0)
+
+
+def test_data_clamp_and_data_stimulate():
+    """In theory people shouldn't use these two together, but at least it shouldn't break."""
+    comp = jx.Compartment()
+    comp.insert(HH())
+    comp.record()
+    clamp = -50.0 * jnp.ones((1000,))
+    stim = 0.1 * jnp.ones((1000,))
+
+    def provide_data(clamp, stim):
+        data_clamps = comp.data_clamp("v", clamp)
+        data_stims = comp.data_stimulate(stim)
+        return data_clamps, data_stims
+
+    def simulate(clamp, stim):
+        data_clamps, data_stims = provide_data(clamp, stim)
+        return jx.integrate(comp, data_clamps=data_clamps, data_stimuli=data_stims)
+
+    jitted_simulate = jax.jit(simulate)
+
+    s = jitted_simulate(clamp, stim)
+    assert np.all(s[:, 1:] == -50.0)
+
+
+def test_data_clamp_and_stimulate():
+    """Test that data clamp overrides a previously set stimulus."""
+    comp = jx.Compartment()
+    comp.insert(HH())
+    comp.record()
+    clamp = -50.0 * jnp.ones((1000,))
+    stim = 0.1 * jnp.ones((800,))
+    t_max = clamp.shape[0] * 0.025  # make sure the stimulus gets padded
+    comp.stimulate(stim)
+
+    def simulate(clamp):
+        data_clamps = comp.data_clamp("v", clamp)  # should override the stimulation
+        return jx.integrate(comp, data_clamps=data_clamps, t_max=t_max)
+
+    jitted_simulate = jax.jit(simulate)
+
+    s = jitted_simulate(clamp)
+    assert np.all(s[:, 1:] == -50.0)
+
+
+def test_data_clamp_and_clamp():
+    """Test that data clamp overrides a previous clamp to the same compartment and can be added to a previous clamp on a different compartment."""
```

Comment:
First line of a docstring should be a single line with less than 88 characters

Reply:
(Actually, I would have hoped that black complains about this, but apparently it doesn't)